tools/mirror/src/lib.rs

```diff
@@ -1638,23 +1766,35 @@ impl<T: ChainAccess> TxMirror<T> {
                 .iter()
                 .map(|s| MappedChunk { shard_id: *s, txs: Vec::new() })
                 .collect();
-            let mut block = MappedBlock { source_height: last_height, chunks };
+            // we set source_hash to 0 because we don't actually care about it here, and it doesn't even exist since these are
+            // not transactions corresponding to some actual block, but just extra txs create account actions in the first few blocks.
+            let mut block = MappedBlock {
+                source_hash: CryptoHash::default(),
+                source_height: last_height,
+                chunks,
+            };
             for h in next_heights {
                 self.add_create_account_txs(h, target_head, &mut tracker, &mut block.chunks)
                     .await?;
             }
             if block.chunks.iter().any(|c| !c.txs.is_empty()) {
                 tracing::debug!(target: "mirror", "sending extra create account transactions for the first {} blocks", CREATE_ACCOUNT_DELTA);
                 tracker.queue_block(block, &self.target_view_client, &self.db).await?;
-                let b = tracker.next_batch(&self.target_view_client, &self.db).await?;
-                self.send_transactions(b).await?;
-                tracker.on_txs_sent(&self.db, target_height).await?;
+                let mut b = tracker.next_batch(&self.target_view_client, &self.db).await?;
+                self.send_transactions(b.chunks.iter_mut().flat_map(|c| c.txs.iter_mut())).await?;
+                tracker
+                    .on_txs_sent(
+                        &self.db,
+                        crate::chain_tracker::SentBatch::MappedBlock(b),
+                        target_height,
+                    )
+                    .await?;
             }
         }
 
         self.queue_txs(&mut tracker, target_head, false).await?;
 
-        self.main_loop(tracker, target_height, target_head).await
+        self.main_loop(tracker, target_height, target_head, source_hash).await
     }
 }
 
```

Comment:
And this file is long, 1800 lines.

Reply:
yea... this whole thing could use a refactor: https://github.com/near/nearcore/blob/67a527bf39534bfbef808f62e71c35eafb13ad0b/tools/mirror/src/chain_tracker.rs#L124